notes-webapp/src/main/webapp/vue-app/notes/components/NotesOverview.vue

```diff
@@ -809,13 +813,17 @@ export default {
       const contentChildren = docElement.getElementsByTagName('body')[0].children;
       const links = docElement.getElementsByTagName('a');
       const tables = docElement.getElementsByTagName('table');
+      docElement.querySelectorAll('pre > code').forEach((e) => codeHighlighter.highlightBlock(e));
```

Comment:
Using a 149KB of new framework + its corresponding CSS seems not relevant. Please try to delete the usage of this framework if it's not a major functional requirement.

```suggestion
```

Reply:
> Using a 149KB of new framework + its corresponding CSS seems not relevant. Please try to delete the usage of this framework if it's not a major functional requirement, at least in "**View Mode**".

This library comes with ckeditor and used in the codesnippet plugin, so i created an amd library of it + a portal skin of the choose style of the code to ensure the WYSIWYG for codesnuppet because there is no other way to do that in view mode. ckidetor.getData() doesn't return the exactly highlighted html code.
https://github.com/Meeds-io/commons/pull/551/commits/90092cc73ff14e478395b4f844cd14ed5c152ddf#diff-60986d51a51cfa41dd1f469bd45a68212c4512d265ab7dfb946a3127b6671862R5

Reply:
ok for Style but no need to make the framework work in **View Mode**
The plugin is intended to generate an HTML snippet that has to be saved as is. And when doing so, no need to retrigger the plugin in View mode to have a good display of the injected HTML.

Reply:
Ok but without the userId the listener is useless
 https://github.com/Meeds-io/notes/pull/843/files#diff-3198ed87ff299273c36b60c054d7cfdef2b2eada4a3f90bd54d51fe498237cb7R61-R62

Reply:
OK but even if we keep the styles, in the view mode we will not have the same display of the code as in edit mode, as all highlighted parts are transformed into span tags with css classes, but these span tags are removed when call editor.getData()

Reply:
And if we save the generated HTML, we will not be able to edit it again.

Reply:
> And if we save the generated HTML, we will not be able to edit it again.

We have to not use `HTMLSanitizer` anymore when writing the information. (we don't have to loose the original information entered by the end user)
But we use `v-sanitized-html` to sanitize the html on rendering in order to preserve the original information.

Reply:
The problem is not in the `HtmlSanitizer` it's in the ckeditor itself, in the `editor.getData()`

Reply:
> And if we save the generated HTML, we will not be able to edit it again.

>The problem is not in the HtmlSanitizer it's in the ckeditor itself, in the editor.getData()

This is may be a cross cutting concern about CKEditor contents. We may need to preserve the HTML as displayed in CKEditor (`CKEDITOR.instances.**INSTANCE_NAME**.document.$.body.innerHTML`) to preserve the WYSIWYG effect, not only for notes. At the same time, we will need to preserve the response for `editor.getData()` to be able to edit the content with CKEditor plugins.

This would be a good alternative to [PHP backend processor](https://ckeditor.com/docs/ckeditor4/latest/features/codesnippetgeshi.html#back-end-configuration) of a version of this plugin.

We will talk more about this in the dedicated meeting.

Reply:
I have already used something similar to preserve the embedded iframe of the media plugin
https://github.com/Meeds-io/notes/pull/843/files#diff-43927e45889ed109fe9f553ea38b02993f822a8fb1827fa9d26c660fa5cf3c79R813-R834
should i try to do the same for highlighter too ?

Reply:
The idea is to minimize the **CPU & Memory consumption**, in browser side, when displaying Notes in Page. It should be as much static as possible to optimize the necessary ressources to display a page with multiple **SNVs** based on Notes.
Having to load a **heavy JS library** in addition to a computing to transform the HTML each time we display the page wouldn't be performant.